web3/contract/base_contract.py

```diff
@@ -747,18 +762,21 @@ def encode_abi(
 
         :param data: defaults to function selector
         """
-        fn_abi, fn_selector, fn_arguments = get_function_info(
+        args = args or tuple()
+        kwargs = kwargs or {}
+
+        fn_info = get_abi_element_info(
```

Comment:
`get_abi_element_info` calls `get_abi_element` internally with the arguments and `cls.abi` to filter the correct function and put it in `fn_info`.

This information is needed for the encode call and only uses the ABI of the function being called, as well as the data being encoded using `selector`, `arguments`, and `data`. The return type of `get_function_info` was originally a `Tuple[ABIFunction, HexStr, Tuple[Any, ...]]`. There was a comment in an earlier thread somewhere to adapt the Tuple into a new type, which became a TypedDict called `ABIElementInfo`.

Maybe I could add a comment or something to clear that up, what do you think?

Reply:
> `get_abi_element_info` calls `get_abi_element` internally with the arguments and `cls.abi` to filter the correct function and put it in `fn_info`.

My biggest concern is that we're using `cls.abi` (the whole abi object) and not `get_all_function_abis(cls.abi)` (only function abis) as the argument here. I think that may be redundant to filter for only function abis, but it just feels like a decent gut check and mmakes reading the code a bit easier. Does that make sense?

Reply:
That's reasonable, I will add that filter!

Reply:
Diving deeper here, `encode_abi` should work for any ABI element, so it's unfortunate the function argument is named `fn_name`. I'm dont think a filter would make it easier to tell what's going on here, ideally we would change the variable names but probably not a good time to touch `encode_abi` just yet. Do you think we should still do the filter, a comment, or something else?

Reply:
It looks like the old logic here didn't actually restrict things to functions, even though the arg names would lead you to believe it. I wouldn't be against changing `fn_name` -> `abi_element_name` and removing leaving the function filtering out of it. Thoughts on that?

Reply:
I like that plan and I will document that change in the migration guide!